ext/django_stubs_ext/aliases.py

```diff
@@ -5,14 +5,16 @@
     from django.utils.functional import _StrOrPromise as StrOrPromise
     from django.utils.functional import _StrPromise as StrPromise
 
+    # Obsolete type aliases. Use the QuerySet class directly instead.
     QuerySetAny = _QuerySetAny
     ValuesQuerySet = _QuerySet[_T, _Row]
 else:
     from django.db.models.query import QuerySet
     from django.utils.functional import Promise as StrPromise
 
+    StrOrPromise = typing.Union[str, StrPromise]
+    # Obsolete type aliases. Use the QuerySet class directly instead.
```

Comment:
```suggestion
    # Deprecated type aliases. Use the QuerySet class directly instead.
```

I think that we should deprecate them in the future with a real warning.